### ISSUE TITLE
Display inline keyboard only to specific user in group

### ISSUE THREAD

**Issue by xiokax** [php-telegram-bot/core#875]
- Operating system: - 
- PHP version: 7.2.7
- PHP Telegram Bot version: 0.54.0
- Using MySQL database: yes 
- MySQL version: 5.7.21
- Update Method: Webhook
- Self-signed certificate: yes
- RAW update (if available): 

### Expected behaviour
When a user invokes /inlinekeyboard command in a group, two buttons should be shown only to the user who invoked the command. On pressing one of the buttons, (1) the buttons should be removed, (2) the bot should send a message to the chat, (3) the query should be handled.

### Actual behaviour
The buttons are shown to all users in the group chat. Everyone can click on the buttons. After clicking on a button, the bot receives the query, but the message (reply) is not sent to the channel, and the buttons are still available. 

### Steps to reproduce
InlinekeyboardCommand.php
```
<?php
/**
 * This file is part of the TelegramBot package.
 *
 * (c) Avtandil Kikabidze aka LONGMAN <akalongman@gmail.com>
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 */
namespace Longman\TelegramBot\Commands\UserCommands;

use Longman\TelegramBot\Request;
use Longman\TelegramBot\Commands\UserCommand;
use Longman\TelegramBot\Entities\InlineKeyboard;

/**
 * User "/inlinekeyboard" command
 *
 * Display an inline keyboard with a few buttons.
 */
class InlinekeyboardCommand extends UserCommand
{

    /**
     *
     * @var string
     */
    protected $name = 'inlinekeyboard';

    /**
     *
     * @var string
     */
    protected $description = 'Show inline keyboard';

    /**
     *
     * @var string
     */
    protected $usage = '/inlinekeyboard';

    /**
     *
     * @var string
     */
    protected $version = '1.0';

    /**
     * Command execute method
     *
     * @return \Longman\TelegramBot\Entities\ServerResponse
     * @throws \Longman\TelegramBot\Exception\TelegramException
     */
    public function execute()
    {
        $chat_id = $this->getMessage()
            ->getChat()
            ->getId();

        $inline_keyboard = new InlineKeyboard([
            [
                'text' => 'Chicken',
                'callback_data' => 'command_chicken'
            ],
            [
                'text' => 'Beef',
                'callback_data' => 'command_beef'
            ]
        ]);
        $inline_keyboard->setSelective(true);
        $inline_keyboard->setOneTimeKeyboard(true);

        $data = [
            'chat_id' => $chat_id,
            'text' => 'Pick your choice',
            'reply_markup' => $inline_keyboard,
            'reply_to_message_id' => $this->getMessage()->getMessageId(),
            'selective' => true
        ];

        return Request::sendMessage($data);
    }
}
```
CallbackqueryCommand.php
```
<?php
/**
 * This file is part of the TelegramBot package.
 *
 * (c) Avtandil Kikabidze aka LONGMAN <akalongman@gmail.com>
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 */
namespace Longman\TelegramBot\Commands\SystemCommands;

use Longman\TelegramBot\Request;
use Longman\TelegramBot\Commands\SystemCommand;

/**
 * Callback query command
 *
 * This command handles all callback queries sent via inline keyboard buttons.
 *
 * @see InlinekeyboardCommand.php
 */
class CallbackqueryCommand extends SystemCommand
{

    /**
     *
     * @var string
     */
    protected $name = 'callbackquery';

    /**
     *
     * @var string
     */
    protected $description = 'Reply to callback query';

    /**
     *
     * @var string
     */
    protected $version = '1.0';

    /**
     * Command execute method
     *
     * @return \Longman\TelegramBot\Entities\ServerResponse
     * @throws \Longman\TelegramBot\Exception\TelegramException
     */
    public function execute()
    {
        if (! $this->getCallbackQuery()) {
            return Request::emptyResponse();
        }

        $chat_id = $this->getCallbackQuery()
            ->getMessage()
            ->getChat()
            ->getId();
        $message_id = $this->getCallbackQuery()->getInlineMessageId();

        // 1. Delete message to hide buttons
        Request::deleteMessage([
            'chat_id' => $chat_id,
            'message_id' => $message_id
        ]);

        // 2. Send text to chat
        $callback_query = $this->getCallbackQuery();
        $callback_query_id = $callback_query->getId();
        $callback_data = $callback_query->getData();

        $data = [
            'text' => 'You picked' . $callback_data,
            'channel_id' => $chat_id
        ];
        Request::sendMessage($data); // Not working :-(

        // 3. Answer callbackquery
        // Debug
        $data = [
            'callback_query_id' => $callback_query_id,
            'cache_time' => 5
        ];

        return Request::answerCallbackQuery($data);
    }
}
```
### Extra details
I am not sure whether I understood the Telegram API correctly, but with the parameters selective(true) and reply_to_message_id(id), it should be possible to display the inline buttons only to the user who invoked the command. But thats not working, buttons are shown to all users in the group.

Moreover, to delete the buttons once the user made his choice, I simply want to delete the message containing the inline buttons. It is my (the bots message) - so no additional rights in the group are required, right?

Furthermore, I dont understand why Request::sendMessage($data);  is not working.

The good news is the query is being handled, i.e., no loading icon is displayed once a user selects a button.

Hope, you can help me. 

Kind regards

**Comment by noplanman** [comment]
The `sendMessage` method has no `selective` parameter, only the keyboard does, which you are setting correctly. The `reply_to_message_id` is what enforces that only the user who requested the keyboard sees it.

> (1) the buttons should be removed

This should be working with your code. Although, you can also just [update your message keyboard](https://core.telegram.org/bots/api#editmessagereplymarkup) and removing it with `'reply_markup' => Keyboard::remove()`.

> (2) the bot should send a message to the chat

The reason your message isn't being sent:
`'channel_id' => $chat_id` should be `'chat_id' => $chat_id`

**Comment by jacklul** [comment]
Not possible with Inline Keyboard.

You gotta implement some kind of lock on your side that allows only the command caller to use the buttons.

**Comment by xiokax** [comment]
> This should be working with your code.

It didnt because I made a mistake.. Fixed it and now its working:

`$message_id = $this->getCallbackQuery()->getMessage()->getMessageId();`

> 'channel_id' => $chat_id should be 'chat_id' => $chat_id

Oh my... thanks. Its working like a charm. Now only the issue remains where the keyboard is shown to all users.

> The sendMessage method has no selective parameter, only the keyboard does, which you are setting correctly.

Yes, I know. Just tested this parameter in different places but still no success. Maybe its not working indeed as @jacklul said. But, according to the Telegram API it should work (see https://core.telegram.org/bots/api#sendmessage). The message has to be a reply (which actually is cause reply_to_message_id is set and 'selective' is set to true inside the InlineKeyboard. But it still does not work. Maybe I have another logical error?

At worst, I implement a lock that allows only the command caller to press the buttons. But it would be nice to have this issue solved. Maybe you guys have another idea...🤔
